Structure_Engine/Query/Construction.cs

```diff
@@ -0,0 +1,111 @@
+ */
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.ComponentModel;
+using BH.oM.Base.Attributes;
+using BH.oM.Physical.Constructions;
+using BH.oM.Structure.SurfaceProperties;
+using BH.oM.Quantities.Attributes;
+using BH.oM.Physical.Materials;
+
+namespace BH.Engine.Structure
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+        [Description("Creates a physical Construction from a structural ISurfaceProperty. Extracts the Structural MaterialFragment and creates a physical material with the same name.")]
+        [Input("surfaceProperty", "Structural surface property to convert.")]
+        [Output("construction", "The physical Construction to be used with ISurface such as Walls and Floors.")]
+        public static Construction IConstruction(this ISurfaceProperty surfaceProperty)
+        {
+            if (surfaceProperty.IsNull())
+                return null;
+            return Construction(surfaceProperty as dynamic);
+        }
+
+        //Write specific methods here if building a Construction from a SurfaceProperty via MaterialComposition is not valid
+        /***************************************************/
+        /**** Fallback Method                           ****/
+        /***************************************************/
+        [Description("Creates a physical Construction from a structural ISurfaceProperty. Extracts the Structural MaterialFragment and creates a physical material with the same name.")]
+        [Input("surfaceProperty", "Structural surface property to convert.")]
+        [Output("construction", "The physical Construction to be used with ISurface such as Walls and Floors.")]
+        public static Construction Construction(this ISurfaceProperty surfaceProperty)
+        {
+            if (surfaceProperty.IsNull())
+                return null;
+            MaterialComposition comp = surfaceProperty.IMaterialComposition();
+            double volume = surfaceProperty.IVolumePerArea();
+            double thickness = surfaceProperty.ITotalThickness();
+            List<oM.Physical.Constructions.Layer> layers = new List<oM.Physical.Constructions.Layer>();
+            for (int i = 0; i < comp.Materials.Count(); i++)
+                layers.Add(new oM.Physical.Constructions.Layer()
+                {Material = comp.Materials[i], Thickness = volume * comp.Ratios[i], Name = comp.Materials[i].Name});
+            if (volume < thickness)
+                layers.Add(new oM.Physical.Constructions.Layer()
+                {Material = null, Thickness = thickness - volume, Name = "void"});
+            return Physical.Create.Construction(surfaceProperty.Name, layers);
+        }
+
+        /***************************************************/
```

Comment:
Are the private methods used anywhere? Can't see a reference?

Reply:
Yep, redundant!